internal/services/network/bastion_host_resource.go

```diff
@@ -66,7 +66,7 @@ func resourceBastionHost() *pluginsdk.Resource {
 			"ip_configuration": {
 				Type:     pluginsdk.TypeList,
 				ForceNew: true,
-				Optional: true,
+				Optional: false,
```

Comment:
instead of changing optional to false we should use `Required:     true`
```suggestion
				Required: true,
```